File: 11.container-with-most-water.py
#
# @lc app=leetcode id=11 lang=python3
#
# [11] Container With Most Water
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        # Two pointers:
        # 1. if height[l] < height[r], inc l
        # 2. else, dec r
        # stop when l >= r 
        l, r = 0, len(height) - 1
        maxArea = 0
        while l < r:
            maxArea = max(maxArea, min(height[l], height[r]) * (r - l))
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return maxArea
        
# @lc code=end

File: 11.container-with-most-water.cpp
/*
 * @lc app=leetcode id=11 lang=cpp
 *
 * [11] Container With Most Water
 */

// @lc code=start
#include <vector>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& height) {
        int s = 0, e = height.size() - 1;
        int maxArea = 0;
        while (s < e) {
            int area = (e - s) * min(height[s], height[e]);
        }        
    }
};
// @lc code=end

